persistence-cassandra/src/main/scala/com/evolutiongaming/kafka/flow/snapshot/SnapshotSchema.scala

```diff
@@ -32,6 +32,21 @@ private[snapshot] object SnapshotSchema {
           |PRIMARY KEY((application_id, group_id, topic, partition, key), offset)
           |)
           |""".stripMargin
+      ).first *>
+      session.execute(
+        """CREATE TABLE IF NOT EXISTS snapshots_v2(
+          |application_id TEXT,
+          |group_id TEXT,
+          |topic TEXT,
+          |partition INT,
+          |key TEXT,
+          |offset BIGINT,
+          |created TIMESTAMP,
+          |metadata TEXT,
+          |value BLOB,
+          |PRIMARY KEY(application_id, group_id, topic, partition, key)
```

Comment:
Seems like it will define partition key as `application_id`. Doesn't it mean that it will put all data on a single node per application?

Reply:
If it was `PRIMARY KEY(application_id)`, it would, indeed put everything on a single node.

Or, to give another example, if you write `PRIMARY KEY(application_id, group_id, topic, partition)`, then everything related to the partition will be on a single node, because other fields in the key will stay the same. Of course, if the partition is the same, but `application_id` are different, then Cassandra will be able to put the data to several nodes for _different_ applications.

In our case, i.e. `PRIMARY KEY(application_id, group_id, topic, partition, key)`, will be able to use several nodes per application, group, topic, partition etc. because we have `key` there.

Everything related to one key will have to go to a single node though.

Reply:
From [Cassandra documentation](https://cassandra.apache.org/doc/latest/cql/ddl.html#partition-key):

> A CQL primary key is composed of 2 parts:

>- the partition key part. It is the first component of the primary key definition. It can be a single column or, using additional parenthesis, can be multiple columns. A table always have at least a partition key, the smallest possible table definition is:
`CREATE TABLE t (k text PRIMARY KEY);`
>- the clustering columns. Those are the columns after the first component of the primary key definition, and the order of those columns define the clustering order.

>Some example of primary key definition are:
`PRIMARY KEY (a)`: a is the partition key and there is no clustering columns.
`PRIMARY KEY (a, b, c)` : a is the partition key and b and c are the clustering columns.
`PRIMARY KEY ((a, b), c)` : a and b compose the partition key (this is often called a composite partition key) and c is the clustering column.
>Within a table, CQL defines the notion of a partition. A partition is simply the set of rows that share the same value for their partition key. 

Given the above, we should read `PRIMARY KEY(application_id, group_id, topic, partition, key)` as "`application_id` is the partition key and `group_id, topic, partition, key` is the sorting key, so we put rows with the same `application_id` in the same partition and on the same node".

Reply:
Yes, you are right. Will fix it.

Reply:
Thanks @rtar 👍

Reply:
@asavelyev01 It is such a pleasure when there is someone to review your code. I would have spent days debugging why everything went to one node suddenly if not your comment.